test/benchmarks/src/main/scala/scala/reflect/NamesBenchmark.scala

```diff
@@ -0,0 +1,155 @@
+      case "WeakAutoTrimConcurrentMapNameTable1" => () => new WeakAutoTrimConcurrentMapNameTable1[AnyRef](new NameBase(_))
+      case "WeakAutoTrimConcurrentMapNameTable2" => () => new WeakAutoTrimConcurrentMapNameTable1[AnyRef](new NameBase(_))
+      case "WeakNoAutoTrimConcurrentMapNameTable1" => () => new WeakNoAutoTrimConcurrentMapNameTable1[AnyRef](new NameBase(_))
+      case "WeakNoAutoTrimConcurrentMapNameTable2" => () => new WeakNoAutoTrimConcurrentMapNameTable2[AnyRef](new NameBase(_))
+      case "StrongConcurrentNodeInterner" => () => new StrongConcurrentNodeInterner(new NameBase(_))
+      case "WeakFixedSizeNoAutoTrimConcurrentNodeInterner" => () => new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_))
+      case "WeakFixedSizeAutoTrimConcurrentNodeInterner" => () => new WeakFixedSizeAutoTrimConcurrentNodeInterner(new NameBase(_))
+      case "Find0" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find(chars, start, count)
+        }
+      case "Find1" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find1(chars, start, count)
+        }
+      case "Find2" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find2(chars, start, count)
+        }
+      case "Find3" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find3(chars, start, count)
+        }
+      case "Safe" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find(chars, start, count)
+        }
+      case "Unsafe" => () =>
+        new UnsafeWeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(chars: Array[Char], start: Int, count: Int) = super.find(chars, start, count)
+        }
+      case "Tail" => () =>
+        new WeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(key: String) = super.find(key)
+        }
+      case "NoTail" => () =>
+        new UnsafeWeakFixedSizeNoAutoTrimConcurrentNodeInterner(new NameBase(_)) {
+          override def find(key: String) = super.findNoTail(key)
+        }
+    }
+    fullNameTable = newNameTable()
+    names foreach {
+      fullNameTable.find(_)
+    }
+  }
+
+  @Benchmark def findOldString(bh: Blackhole): Any = {
+    val nameTable: NameTable[_ <: AnyRef] = fullNameTable
+    val res = new Array[AnyRef](arraySize)
+    var i = 0
+    while (i < arraySize) {
+      res(i) = nameTable.find(names(i))
+      i += 1
+    }
+    bh.consume(res)
```

Comment:
The size parameter is being used to change both the size of the pre-populated name table and the number of lookups done from within it. I'd prefer to parameterize just the former, and measure a single lookup (perhaps drawn at random from the sequence of known names).